src/common/ngx_wasm_socket_tcp.c

```diff
@@ -284,7 +319,15 @@ ngx_wasm_socket_tcp_connect(ngx_wasm_socket_tcp_t *sock)
                        "wasm tcp socket no resolving: %V",
                        &sock->resolved.host);
 
-        return ngx_wasm_socket_tcp_connect_peer(sock);
+        rc = ngx_wasm_socket_tcp_connect_peer(sock);
+        if (sock->connected) {
+#if (NGX_SSL)
+            if (sock->ssl_conf) {
+                return ngx_wasm_socket_tcp_ssl_handshake(sock);
```

Comment:
Did you find some kind of problem with SSL handshakes? This sort of catches my eye... And now I'm wondering if we don't have some problems with triggering SSL handshakes?

Reply:
Yes I did.

I noticed the SSL handshake was not being performed when the dispatched request targeted a unix domain socket.

And that was happening because when the request was dispatched, `ngx_wasm_socket_tcp_connect` ran only once due to `ngx_wasm_socket_tcp_connect_peer` returning `NGX_OK` rather than `NGX_AGAIN`.

This prevented `ngx_wasm_socket_tcp_ssl_handshake` from ever being invoked [here](https://github.com/Kong/ngx_wasm_module/blob/main/src/common/ngx_wasm_socket_tcp.c#L215-L222) since the first time `ngx_wasm_socket_tcp_connect` is called `sock->connected` would still be false.

I was indeed going to ping you over this change as I'm not sure this is the solution we want for this behavior.

Reply:
Hmm so this means when specifying an IP address (not `unix`, and no DNS resolution) we are also not initiating an SSL handshake... We will need tests for this use-case as well!